src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,132 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        Node<T> node = head;
```

Comment:
optional: for more productivity we can add checking index < size / 2 and index > size / 2